databroker/src/grpc/kuksa_val_v2/val.rs

```diff
@@ -39,22 +39,96 @@ const MAX_REQUEST_PATH_LENGTH: usize = 1000;
 impl proto::val_server::Val for broker::DataBroker {
     async fn get_value(
         &self,
-        _request: tonic::Request<proto::GetValueRequest>,
+        request: tonic::Request<proto::GetValueRequest>,
     ) -> Result<tonic::Response<proto::GetValueResponse>, tonic::Status> {
-        Err(tonic::Status::new(
-            tonic::Code::Unimplemented,
-            "Unimplemented",
-        ))
+        debug!(?request);
+        let permissions = match request.extensions().get::<Permissions>() {
+            Some(permissions) => {
+                debug!(?permissions);
+                permissions.clone()
+            }
+            None => return Err(tonic::Status::unauthenticated("Unauthenticated")),
+        };
+
+        let broker = self.authorized_access(&permissions);
+
+        let request = request.into_inner();
+
+        let signal_id = match get_signal(request.signal_id, &broker).await {
+            Ok(signal_id) => signal_id,
+            Err(err) => return Err(err),
+        };
+
+        let datapoint = match broker.get_datapoint(signal_id).await {
+            Ok(datapoint) => datapoint,
+            Err(ReadError::NotFound) => return Err(tonic::Status::not_found("Path not found")),
+            Err(ReadError::PermissionDenied) => {
+                return Err(tonic::Status::permission_denied("Permission denied"))
+            }
+            Err(ReadError::PermissionExpired) => {
+                return Err(tonic::Status::unauthenticated("Unauthenticated"))
```

Comment:
Wouldn't it be beneficial if we tell them in the message "Unauthenticated because your token expired"? 

```
 Err(ReadError::PermissionExpired) => {
                    return Err(tonic::Status::unauthenticated(format!(
                        "Permission expired (id: {})",
                        signal_id
                    )))
```
like here in line 120 ff.

Reply:
Makes sens to say that auth expired. 

There are theoretically a lot of GRPC error codes (see for example https://grpc.github.io/grpc/cpp/md_doc_statuscodes.html) that we cannot control but the client still can receive. But it might be a good policy to mention all that we explicitly return in our implementation